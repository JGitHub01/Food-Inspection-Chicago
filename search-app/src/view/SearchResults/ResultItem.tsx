import moment from 'moment';
import {IResultItem} from 'lib/models';

export default function ResultItem(props: IResultItem) {
  const { inspection_date, dba_name, address, risk } = props;

  return (
    <div className='result-item' data-testid='result-item'>
      <div className='result-item__row result-item__date-row'>
        <span>{moment(inspection_date).format('MMM Do, YYYY')}</span>
        <span className='business-name'>{dba_name}</span>
      </div>
      <div className='result-item__row'>
        {address}
      </div>
      <div className='result-item__row'>
        {risk}
      </div>
    </div>
  );
}
