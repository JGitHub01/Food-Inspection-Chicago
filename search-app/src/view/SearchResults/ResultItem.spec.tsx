import { render } from '@testing-library/react';
import ResultItem from './ResultItem';

describe('ResultItem', () => {
  const item = {
    "inspection_date": "2021-05-13T00:00:00.000",
    "dba_name": "SCHUBA'S TAVERN/TIED HOUSE",
    "risk": "Risk 1 (High)",
    "address": "3153-3159 N SOUTHPORT AVE "
  }
  it('should render the basic snapshot view', () => {
    const { getByTestId } = render(<ResultItem {...item}/>);
    const resultItem = getByTestId('result-item');
    expect(resultItem).toMatchSnapshot();
  });
});
