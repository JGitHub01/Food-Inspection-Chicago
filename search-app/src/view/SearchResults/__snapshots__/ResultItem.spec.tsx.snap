// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultItem should render the basic snapshot view 1`] = `
<div
  class="result-item"
  data-testid="result-item"
>
  <div
    class="result-item__row result-item__date-row"
  >
    <span>
      May 13th, 2021
    </span>
    <span
      class="business-name"
    >
      SCHUBA'S TAVERN/TIED HOUSE
    </span>
  </div>
  <div
    class="result-item__row"
  >
    3153-3159 N SOUTHPORT AVE 
  </div>
  <div
    class="result-item__row"
  >
    Risk 1 (High)
  </div>
</div>
`;
